### PR TITLE
Add examples to Module.init docstring

### PR DESCRIPTION
**Pull request by zaxtax**
This improves the docstring on `Module.init` addressing #2653

### CODE REVIEW

File: flax/linen/module.py [hunk 1]
```diff
@@ -1488,6 +1488,29 @@ def init(self,
...
+      from jax import random
+      import jax.numpy as jnp
+      import flax.linen as nn
+
+      model = nn.Dense(features=5)
+      rng, key = random.split(random.PRNGKey(0))
+      x = random.normal(key, (10,))
+      params = model.init(key, x)
+
+    Alternatively, if named collections are used. Example::
```
Comment (8bitmp3) [inline flax/linen/module.py:1502]: Suggestion:

- The first sentence may read like it's not finished. You could rephrase it to imply that this is a second/alternative example. For example"

```suggestion
    Example (with named collections)::
```

File: flax/linen/module.py [hunk 2]
```diff
@@ -1496,7 +1519,8 @@ def init(self,
       jit_init(rng, jnp.ones(input_shape, jnp.float32))
 
     Args:
-      rngs: The rngs for the variable collections.
+      rngs: The rngs for the variable collections. This argument will also
+        accept a single rng key as well.
```
Comment (8bitmp3) [inline flax/linen/module.py:1523]: Nit: since it's an acronym

```suggestion
      rngs: The RNGs for the variable collections. This argument will also
        accept a single RNG key as well.
```

(or "PRNGs")

**Comment by zaxtax** [comment]
Great suggestions!

**Comment by marcvanzee** [comment]
Duplicate of #2788
